compiler/src/parsing/driver.re

```diff
@@ -69,3 +69,74 @@ let parse = (~name=?, lexbuf): Parsetree.parsed_program => {
   | parses => raise(Error(startpos, AmbiguousParse(List.map(fst, parses))))
   };
 };
+
+let scan_for_imports = (~defer_errors=true, filename: string): list(string) => {
+  let ic = open_in(filename);
+  let lexbuf = Lexing.from_channel(ic);
+  try({
+    let {Parsetree.comments, Parsetree.statements} = parse(lexbuf);
+    let implicit_opens =
+      switch (comments) {
+      | [Block({cmt_content, cmt_loc}), ..._]
+          when
+            Str.string_match(
+              Str.regexp_string("grainc-flags"),
+              cmt_content,
+              0,
+            ) =>
+        // [FIXME] Kind of a gross hack. We should have less hard-coding.
+        let maybe_pervasives =
+          if (Str.string_match(
+                Str.regexp_string("grainc-flags --compilation-mode=runtime"),
```

Comment:
Since you already check that the string starts with `grainc-flags`, here we probably should just check that the string contains `--compilation-mode=runtime` or `--no-pervasives` since there can be multiple flags supplied.

Reply:
We actually should use Cmdliner to parse this, since it also allows various forms of the flags that would be too annoying to all handle here.